src/openassetio-python/package/openassetio/pluginSystem/PythonPluginSystem.py

```diff
@@ -92,29 +86,79 @@ def scan(self, paths):
+                            f"contianing __init__.py {itemPath}"
                         )
-                        self.__logger.log(self.__logger.Severity.kDebug, msg)
                         continue
                 else:
                     # Its a file, check if it is a .py/.pyc module
                     _, ext = os.path.splitext(itemPath)
                     if ext not in self.__validModuleExtensions:
-                        msg = (
-                            "PythonPluginSystem: Ignoring as its not a python module %s" % itemPath
+                        self.__logger.debug(
+                            f"PythonPluginSystem: Ignoring as its not a python module {itemPath}"
                         )
-                        self.__logger.log(self.__logger.Severity.kDebug, msg)
                         continue
 
-                self.__logger.log(
-                    self.__logger.Severity.kDebug,
-                    "PythonPluginSystem: Attempting to load %s" % itemPath,
-                )
+                self.__logger.debug(f"PythonPluginSystem: Attempting to load {itemPath}")
 
                 self.__load(itemPath)
 
+    def scan_entry_points(self, entryPointName):
+        """
+        Searches packages for entry points that define a
+        PythonPluginSystemPlugin through a top-level `plugin` variable.
+
+        @note The order of discovery is determined by `importlib`, only
+        the first plugin with any given identifier will be registered.
+
+        @param entryPointName `str` The entry point name to search for
+        (see: importlib_metadata.entry_points group).
+
+        @returns True if entry point discovery is possible, False if
+        there was a problem loading importlib_metadata.
+        """
+
+        # We opt to use the backport implementation of modern importlib to avoid
+        # needing to support 3+ code paths to cover Python 3.7 to 3.10. It is
+        # made available for 3.7 onwards (for now, at least).
+        # Pip installs should have this module available, but other methods may not,
+        # so be tolerant of it being missing.
+        try:
+            import importlib_metadata  # pylint: disable=import-outside-toplevel
+        except ImportError:
+            self.__logger.warning(
+                "PythonPluginSystem: Can not load entry point plugins as the importlib_metadata "
+                "package is unavailable."
+            )
+            return False
+
+        self.__logger.debug(
+            f"PythonPluginSystem: Searching packages for '{entryPointName}' entry points."
+        )
+
+        for entryPoint in importlib_metadata.entry_points(group=entryPointName):
+
+            self.__logger.debug(f"PythonPluginSystem: Found entry point in {entryPoint.name}")
+            try:
+                module = entryPoint.load()
+            except Exception as ex:  # pylint: disable=broad-except
+                self.__logger.error(
```

Comment:
Yeah, it would be better wouldn't it.  (Don't forget that `self.__logger` isn't a python logger, its a `LoggerInterface` logger, so you'd need to grab the lines from `format_traceback` or similar, but not a big deal. 

I think we did it this way before with some mild notion that log entries were single-line, but that doesn't really need to be the case I dont think?

Reply:
:man_facepalming: of course it's not a normal Python logger, I wasn't thinking.  Yeah I think a full stack trace is really useful when debugging.

Reply:
Full tb printed [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/768/commits/020d67c7411a179dfc57d35c65b30bc5fdf13213).